SpatialGDK/Source/SpatialGDK/Public/Interop/ActorSystem.h

```diff
@@ -163,13 +171,19 @@ class ActorSystem
 	void SendRemoveComponents(Worker_EntityId EntityId, TArray<Worker_ComponentId> ComponentIds) const;
 
 	const FSubView* ActorSubView;
+	const FSubView* AuthoritySubView;
```

Comment:
Given that they can never be null and are initialized as references already, why not store them as references too?

Reply:
Initially I've implemented them as references, too, but changed to pointers at @samiwh's note that they won't be assignable in this case.

Reply:
Interesting - is there any particular reason why the ActorSystem needs to be movable? I can only see it created as a unique pointer which automatically makes the resulting unique pointer movable even if the class itself wasn't.

In terms of expectations, when I say a pointer member like this I'm always asking myself "could this be null" when accessing it. In a way, the pointer type encodes the possibility of it being null similar to how an optional does it. In contrast, references can safely be assumed to never be null and don't need a null check before accessing them.

In normal C++, which might not apply to Unreal C++, there actually is a way out of this to make classes movable that otherwise wouldn't because they have reference member: You can use [`std::reference_wrapper`](https://en.cppreference.com/w/cpp/utility/functional/reference_wrapper) to encode in the type that this is a reference (and will never be null) but that the owning object can still be moved. But as mentioned, I would first ask myself if the type really needs to be movable at all.

Reply:
Broad reason I asked - a lot of things become desirable to move when they either become testable or we move away from dumping all types into a single place (net driver). And I find the mental overhead of working out if X type is built in some "irregular" way because it does something clever, or because it just happened to work, or because the author made a mistake that just didn't break, greater than the overhead of manging these types of raw pointers. Depending on an object you don't own doesn't normally indicate that something can't move so on seeing an unmoveable class it's not my first guess. However this is largely a matter of setting expectation in gdk type design and refracting the parts of the code base that have no consistent structure.

Other places this kind of broad request has come up is in e.g. shared pointer. However that's probably less controversial as there isn't really a trade off being made for different kinds of safety there.

It's possible actor system will never be something that we want to move and reference wrapper might be a good compromise. Also happy to take this offline. This specific instance doesn't matter that much.

I personally find std::reference_wrapper a bit weird to look at although I can't think of any reason I don't like it other than I don't normally see it outside generic code.